File: backend/src/main/java/com/profroid/profroidapp/jobssubdomain/presentationLayer/JobResponseModel.java
package com.profroid.profroidapp.jobssubdomain.presentationLayer;

import com.profroid.profroidapp.jobssubdomain.dataAccessLayer.JobType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@AllArgsConstructor
public class JobResponseModel {

    private String jobId;

    private String jobName;
    private String jobDescription;

    private Double hourlyRate;
    private Integer estimatedDurationMinutes;

    private JobType jobType;

    private boolean isActive;
}

File: frontend/src/features/jobs/models/JobResponseModel.ts
import type { JobType } from './JobType';

export interface JobResponseModel {
  jobId: string;

  jobName: string;
  jobDescription: string;
  jobNameFr?: string;
  jobDescriptionFr?: string;

  hourlyRate: number;
  estimatedDurationMinutes: number;

  jobType: JobType;

  active: boolean;

  imageFileId?: string;
}
